compiler/packages/babel-plugin-react-compiler/src/HIR/Environment.ts

```diff
@@ -239,7 +239,12 @@ const EnvironmentConfigSchema = z.object({
    * the dependency.
    */
   enableOptionalDependencies: z.boolean().default(true),
-
+  
+  /**
+   * Enables inference of effect dependencies. Still experimental.
+   */
+  EXPERIMENTAL_inferEffectDependencies: z.boolean().default(false),
```

Comment:
nit: we haven't been doing "experimental" etc for feature flags. If it's off by default, it's experimental.